### PR TITLE
Minimap: Add new HUD flag for minimap radar mode

### PR DESCRIPTION
**Pull request by paramat**
Flag default is true to not change default behaviour.
The existing minimap HUD flag remains the master control for minimap.
With radar disabled minimap cycles through 'surface' modes only.
//////////////

### CODE REVIEW

**Comment by bigfoot547** [comment]
You really like adding HUD flags, don'cha? :smile:
:+1:

**Comment by paramat** [comment]
See https://github.com/minetest/minetest_game/pull/1853 for discussion of why this is useful.

File: src/game.cpp [hunk 1]
```diff
@@ -2831,7 +2831,10 @@ void Game::toggleMinimap(bool shift_pressed)
 	MinimapMode mode = MINIMAP_MODE_OFF;
 	if (hud_flags & HUD_FLAG_MINIMAP_VISIBLE) {
 		mode = mapper->getMinimapMode();
-		mode = (MinimapMode)((int)mode + 1);
+		mode = (!(hud_flags & HUD_FLAG_MINIMAP_RADAR) &&
```
Comment (rubenwardy) [inline src/game.cpp]: this is unreadable. I suggest formatting it like so:

```cpp
bool isRadarAllowed = hud_flags & HUD_FLAG_MINIMAP_RADAR;

int nextModeID = (!isRadarAllowed && mode == MINIMAP_MODE_SURFACEx4)
     ? (int)mode + 4
     : (int)mode + 1;

mode = (MinimapMode)nextModeID;
```

File: src/player.cpp [hunk 2]
```diff
@@ -70,7 +70,8 @@ Player::Player(const char *name, IItemDefManager *idef):
 	hud_flags =
 		HUD_FLAG_HOTBAR_VISIBLE    | HUD_FLAG_HEALTHBAR_VISIBLE |
 		HUD_FLAG_CROSSHAIR_VISIBLE | HUD_FLAG_WIELDITEM_VISIBLE |
-		HUD_FLAG_BREATHBAR_VISIBLE | HUD_FLAG_MINIMAP_VISIBLE;
+		HUD_FLAG_BREATHBAR_VISIBLE | HUD_FLAG_MINIMAP_VISIBLE |
```
Comment (rubenwardy) [inline src/player.cpp]: please align the second bar

**Comment by rubenwardy** [comment]
What happens if a player has radar disabled whilst they're already in radar mode?

File: src/game.cpp (hunk 1 above)
Comment (paramat) [inline src/game.cpp]: Ok.

File: src/player.cpp (hunk 2 above)
Comment (paramat) [inline src/player.cpp]: Ok.

**Comment by paramat** [comment]
Will check.

File: src/game.cpp (hunk 1 above)
Comment (nerzhul) [inline src/game.cpp]: bool isRadarAllowed = hud_flags & HUD_FLAG_MINIMAP_RADAR == 0;

it's more accurate and recommended by checkstyle tools

Comment (paramat) [inline src/game.cpp]: No need now due to new simpler implementation.

File: src/player.cpp (hunk 2 above)
Comment (paramat) [inline src/player.cpp]: Done.

**Comment by paramat** [comment]
Updated and retested.
I improved the HUD flag name and the lua name, both for consistency.
I now use simpler code that copes with disabling radar mode while in radar mode, the next press of F9 goes straight to 'minimap mode off' (tested).
Thanks for the review this was a little sloppy in a few ways.

**Comment by paramat** [comment]
Support from SmallJoker (Krock) http://irc.minetest.net/minetest-dev/2017-08-19#i_5048488
Support from sfan5 http://irc.minetest.net/minetest-dev/2017-08-19#i_5048516 but i have addressed his point.

**Comment by paramat** [comment]
LINT is complaining about 2 newlines used in l_client.cpp, ignore, not part of this PR.

**Comment by paramat** [comment]
http://irc.minetest.net/minetest-dev/2017-08-19#i_5048982
> have you checked that setting the flag when radar is open closes it?

I found this will be difficult and more complex to do, but i have managed to make radar updates freeze if radar is disabled while in radar mode by adding code in clientpackethandler.cpp, and the next F9 press cycles straight into surface mode.
So the possible cheat of a player simply staying in radar mode when it is disabled is avoided.

**Comment by paramat** [comment]
However, i will try to improve this further.

**Comment by paramat** [comment]
Solved, see changes in clientpackethandler.cpp. Similar code is already used to force-close minimap when disabled, i used similar code to switch to surface mode when radar is disabled.

**Comment by paramat** [comment]
7657fe7a505425e82d9cab7ae5638f7927138fb0

**Pull request closed without merging**